tesseract_environment/include/tesseract_environment/environment.h

```diff
@@ -531,11 +531,20 @@ class Environment
   mutable std::unordered_map<std::string, tesseract_kinematics::JointGroup::UPtr> joint_group_cache_;
   mutable std::shared_mutex joint_group_cache_mutex_;
 
+  struct kinematic_group_pair_hash
+  {
+      std::size_t operator () (const std::pair<std::string,std::string> &p) const
+      {
+          std::string combined = p.first + p.second;
+          return std::hash<std::string>{}(combined);
+      }
+  };
```

Comment:
This should not be required since;  did it complain without it?

Reply:
Yes, because there is no default hashing for a std::pair, which is required for an unordered_map

Reply:
See comment above about switching cache type.